### ISSUE TITLE
Can't change computed value of useTransition synchronously when it's disabled

### ISSUE THREAD

**Issue by zkulbeda** [vueuse/vueuse#2495]
### Describe the bug

#364 says, that `disabled` option is used for changing the value without transitioning. But, after changing `disabled` to `true`, then changing original value, and then  `disabled` to `false`, the value of `useTransition` won't be equal to original and will make the transition. 

This behavior is necessary to [make a transition from initial value to another value](https://stackblitz.com/edit/vitejs-vite-1dvhai/?file=src%2FApp.vue), where the initial value isn't make a transition with old value.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-tyarou/?file=src%2FApp.vue

### System Info

```Shell
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Memory: 7.68 GB / 19.81 GB
  Binaries:
    Node: 19.0.1 - C:\Program Files\nodejs\node.EXE
    npm: 8.3.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (107.0.1418.62)
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by syy11cn** [comment]
Hi, @zkulbeda. I don't think it's a bug.

### Reason

I copied your code and modified the `set4` function. Also, I added some logs to `computed` and `watch` callbacks.

```diff
function set4() {
  disabled.value = true
+  console.log('set disabled true', disabled.value)
  original.value = 4
  // console.log(current.value);
  disabled.value = false
+  console.log('set disabled false', disabled.value)
}
```

My actions was just click the button to trigger `set4` function. 

![20221203140745](https://user-images.githubusercontent.com/57290456/205427285-c8e4ab44-1cd6-44c8-9a48-b9487248da0f.gif)

Let's see the output.

![image](https://user-images.githubusercontent.com/57290456/205427304-9ca6abb6-6036-4beb-872c-6589652fe557.png)

As you can see, `disabled` is set back to `false` before `useTransition` watches there is some changes to `origin`.

### My suggestion

Just set `disabled` every time before you modify the `origin`, and there's no need to set it back.

https://stackblitz.com/edit/vitejs-vite-6wrvgi/?file=src%2FApp.vue

**Comment by zkulbeda** [comment]
@syy11cn, thank you for response! My actual goal is to make the transition from one value to another. `disabled` is used for setting initial value that shouldn't be transitioned. The problem isn't with `disabled` reference, but with `original.value`!=`current.value` after enabling the transition. It's my fault I chose the wrong example.  See [this example](https://stackblitz.com/edit/vitejs-vite-1dvhai/?file=src%2FApp.vue). And also I changed the first example to make it clear. 

The last test in [useTransition/index.test.ts](https://github.com/vueuse/vueuse/blob/main/packages/core/useTransition/index.test.ts#L210) says that this can be done with synchronous changes, but it isn't. So I suppose this can be considered as a bug.
